### ISSUE TITLE
Strange Excel Validator behavior

### ISSUE THREAD

**Issue by ASamSam** [OfficeDev/office-js#1804]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

I meet strange Excel validator behavoir. My goal is to forbid user to enter some special symbols like '*'. so i found a solution based on ROW and INDIRECT.
so i do create some validators with javascript (from Excel addin)
then when i`m trying to test - it woks only after manual access to Data Validation dialog!

## Expected Behavior
Validator starts to work correctly just after adding.

## Current Behavior
the expected behavior appears only after manual accessing the data validation dialog

## Steps to Reproduce, or Live Example
execute script, click button - puts excel validators
try to enter in A1: 'aa*' - allowed
put 'aa' to A1
open 'data validation' dialog for A1 to check if everything is correct
click OK (nothing changed)
try to enter in A1: 'aa*' - forbidden with Validator!

* Link to live example: [https://gist.github.com/ASamSam/23e0a8be95582ab66322cbebb11f30f9](https://gist.github.com/ASamSam/23e0a8be95582ab66322cbebb11f30f9)

## Context
I`m not able to use the validator in Excel

## Your Environment
Same behavior in excel online and classic one

link on stackoverflow:
[https://stackoverflow.com/questions/67126662/strange-excel-validator-behavior?noredirect=1#comment118664152_67126662](https://stackoverflow.com/questions/67126662/strange-excel-validator-behavior?noredirect=1#comment118664152_67126662)

**Comment by jipinshi** [comment]
I can repro the issue and will investigate.

**Comment by jipinshi** [comment]
5036128 was created

**Comment by jipinshi** [comment]
Thanks reporting the issue. One way could do similar thing is to hook onDataChangeEvent and then in the callback to use regex to check the new value, if the value breaks the rule, then can use the oldValue to recover. @ASamSam
